### PR TITLE
Better err msg when @ann is on wrong type of node 

### PR DESCRIPTION
**Pull request by vmunishwar**
#325

### CODE REVIEW

File: pkg/workspace/doc_extractor.go [hunk 1]
```diff
@@ -89,8 +89,8 @@ func (v DocExtractor) checkNonDocs(val interface{}, annName template.AnnotationN
 				// TODO check for annotation emptiness
 				_, isDoc := node.(*yamlmeta.Document)
 				if !isDoc {
-					errMsg := "Expected YAML document to be annotated with %s but was %T"
-					return fmt.Errorf(errMsg, annName, node)
+					errMsg := "Found @%s on %s (%s:%d); only Documents (---) can be annotated with @%s"
+					return fmt.Errorf(errMsg, annName, node.UserFriendlyName(), node.GetPosition().GetFile(), node.GetPosition().LineNum(), annName)
```
Comment (cppforlife) [inline pkg/workspace/doc_extractor.go]: take a look at https://github.com/vmware-tanzu/carvel-ytt/blob/develop/pkg/yttlibrary/overlay/op.go#L86 for how we typically stringify positions.

Comment (vmunishwar) [reply pkg/workspace/doc_extractor.go]: Thank you. Found a method AsCompactString to use. I have updated the code.

File: pkg/cmd/template/cmd_data_values_test.go [hunk 2]
```diff
@@ -442,7 +442,8 @@ str: str`)
 
 	out := opts.RunWithFiles(cmdtpl.Input{Files: filesToProcess}, ui)
 
-	expectedErrorMessage := "Expected YAML document to be annotated with data/values but was *yamlmeta.MapItem"
+	expectedErrorMessage := "Found @data/values on Map Item (data.yml:4); only Documents (---) can be annotated with @data/values"
+
```
Comment (pivotaljohn) [inline pkg/cmd/template/cmd_data_values_test.go]:
```suggestion
	expectedErrorMessage := "Found @data/values on map item (data.yml:4); only documents (---) can be annotated with @data/values"

```

Let's make node types lowercase as they are ordinary (not proper) nouns.

File: pkg/yamlmeta/ast.go [hunk 3]
```diff
@@ -27,7 +27,10 @@ type Node interface {
 
 	Check() TypeCheck
 
+	UserFriendlyName() string
```
Comment (pivotaljohn) [inline pkg/yamlmeta/ast.go]:
```suggestion
	DisplayName() string
```

Played with this together: is `DisplayName` a better name for this method?

**Review by pivotaljohn — CHANGES_REQUESTED** [review]
Just a few renames and this will be good to go!

File: pkg/cmd/template/cmd_data_values_test.go (hunk 2 above)
Comment (vmunishwar) [reply pkg/cmd/template/cmd_data_values_test.go]: Thank you. I have made the changes.

File: pkg/yamlmeta/ast.go (hunk 3 above)
Comment (vmunishwar) [reply pkg/yamlmeta/ast.go]: Done. Thank you.

**Review by pivotaljohn — APPROVED** [review]
Congratulations on your first merged contribution to `ytt`, @vmunishwar!

This looks great. Every internal detail we can convert into something more intuitive (like referring to a thing as a "map item" rather than a "yamlmeta.MapItem") makes `ytt` _that_ much friendlier and easier to use. 👍🏻

**Pull request merged**